### PR TITLE
Move the @userplot recipes

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
https://github.com/JuliaPlots/RecipesBase.jl/pull/16

### CODE REVIEW

File: src/backends/glvisualize.jl [hunk 1]
```diff
@@ -826,7 +826,7 @@ function gl_bar(d, kw_args)
     hw = if bw == nothing
         mean(diff(x))
     else
-        Float64[cycle(bw,i)*0.5 for i=1:length(x)]
+        Float64[_cycle(bw,i)*0.5 for i=1:length(x)]
```
Comment (mkborregaard) [inline src/backends/glvisualize.jl]: WRT to the history it would look a lot clearer if the _cycle change was separate from the movement of the functions.

Comment (ChrisRackauckas) [inline src/backends/glvisualize.jl]: Alright sounds good

Comment (mkborregaard) [inline src/backends/glvisualize.jl]: 👍 I guess the PR also needs a rebase before eventually merging to sort out the conflicts

Comment (ChrisRackauckas) [inline src/backends/glvisualize.jl]: I'll cherry pick for the cycle change and the `@userplot` change, and close this as a messy pre-PR.

Comment (mkborregaard) [inline src/backends/glvisualize.jl]: Perfect

**Pull request closed without merging**